include/eve/algo/container/soa_vector2.hpp

```diff
@@ -0,0 +1,371 @@
+
+    //==============================================================================================
+    //! @name Modifiers
+    //! @{
+    //==============================================================================================
+    //! @brief Clear the contents of the container.
+    void clear() { size_ = 0; }
+
+    //! @brief Removes an element from the container.
+    //! Has the same invalidation semantics as std::vector.
+    //! end() iterator is not a valid pos.
+    iterator erase(const_iterator pos) { return erase(pos, pos+1); }
+
+    //! @brief Removes the elements in the range [first, last)
+    //! Empty range is OK, does nothing
+    iterator erase(const_iterator f, const_iterator l)
+    {
+      EVE_ASSERT(!empty(), "soa_vector::erase() was called on empty container");
+
+      std::ptrdiff_t distance_f = f - cbegin();
+      std::ptrdiff_t distance_l = l - cbegin();
+      std::ptrdiff_t sz         = distance_l - distance_f;
+
+      eve::algo::copy_backward( algo::as_range(l, cend())
+                              , algo::as_range(begin() + distance_l - sz, end() - sz)
+                              );
+      size_ -= sz;
+      return begin() + distance_f;
+    }
+
+    //! @brief Appends the given element value to the end of the container.
+    //! If the new size() is greater than capacity() then all iterators and references (including
+    //! the past-the-end iterator) are invalidated. Otherwise only the past-the-end iterator is
+    //! invalidated.
+    //! @param value [Value](@eve::vectorizable) to append
+    EVE_FORCEINLINE void push_back(value_type const& value) noexcept
+    {
+      if(size_ == capacity())
+      {
+        // Grow twice per push_back
+        grow(2*size_+1, size());
+      }
+
+      eve::write(value, begin()+size_);
+      size_++;
+    }
+
+    //! @brief Removes the last element of the container.
+    //! Calling pop_back on an empty container results in undefined behavior.
+    //! Iterators and references to the last element, as well as the end() iterator, are invalidated.
+    EVE_FORCEINLINE void pop_back() noexcept
+    {
+      EVE_ASSERT(!empty(), "soa_vector::pop_back() was called on empty container");
+      size_--;
+    }
+
+    //! @brief Resizes the container to contain count elements.
+    //!
+    //! If the current size is greater than `n`, the container is reduced to its first `n` elements.
+    //! If the current size is less than `n`, additional elements are appended. Those elements are
+    //! either default-inserted or copied if `value` is provided.
+    //! @param n      new size of the container
+    //! @param value  the value to initialize the new elements with
+    void resize(std::size_t n, value_type value = {})
+    {
+      auto sz = size_;
+      if(n > capacity())
+      {
+        // Grow twice per resize
+        grow(2*n, size());
+      }
+
+      // If needed, fill with new data
+      if(n > sz) fill(sz, n, value );
```

Comment:
Why not fill up to capacity? I bet it's faster most of the time due to not having to do a masked write.